src/squidpy/exp/im/_detect_tissue.py

```diff
@@ -0,0 +1,562 @@
+
+
+@dataclass(slots=True)
+class BackgroundDetectionParams:
+    """
+    Which corners are background, and how large the corner boxes should be.
+    If no corners are flagged True, the orientation falls back to 'bright background'.
+    """
+
+    ymin_xmin_is_bg: bool = True
+    ymax_xmin_is_bg: bool = True
+    ymin_xmax_is_bg: bool = True
+    ymax_xmax_is_bg: bool = True
+    corner_size_pct: float = 0.01  # size of each corner box as fraction of img height/width
+
+    @property
+    def any_corner(self) -> bool:
+        return any((self.ymin_xmin_is_bg, self.ymax_xmin_is_bg, self.ymin_xmax_is_bg, self.ymax_xmax_is_bg))
+
+
+@dataclass(slots=True)
+class FelzenszwalbParams:
+    """
+    Size-aware superpixel defaults for felzenszwalb segmentation.
+    """
+
+    grid_rows: int = 100  # ~desired grid, only for scale heuristics
+    grid_cols: int = 100
+    sigma_frac: float = 0.008  # blur = this * short side (clipped to [1,5] px)
+    scale_coef: float = 0.25  # scale = coef * target_area
+    min_size_coef: float = 0.20  # min_size = coef * target_area
+
+
+def detect_tissue(
+    sdata: sd.SpatialData,
+    image_key: str,
+    scale: str = "auto",
+    method: DETECT_TISSUE_METHOD | str = DETECT_TISSUE_METHOD.OTSU,
+    corners_are_background: bool = True,
+    close_holes_smaller_than_frac: float = 0.0001,
+    mask_smoothing_cycles: int = 0,
+    new_labels_key: str | None = None,
+    inplace: bool = True,
+    **kwargs: Any,
+) -> xr.DataArray:
+    """
+    Detect tissue and return a boolean mask (y,x) where True = specimen.
+
+    Parameters
+    ----------
+    sdata : sd.SpatialData
+        SpatialData object containing the image
+    image_key : str
+        Key of the image in sdata.images
+    scale : str, default "auto"
+        - If a specific scale key is provided (e.g., "scale0", "scale1", ...),
+          that image scale is used verbatim.
+        - If "auto": pick the smallest available scale. If that smallest scale
+          exceeds the pixel threshold, it is further downscaled to be under the
+          threshold for calculations.
+    method : str or DETECT_TISSUE_METHOD, default OTSU
+        Method to use ("otsu" or "felzenszwalb")
+    corners_are_background : bool, default True
+        Whether all corners should be treated as background
+    close_holes_smaller_than_frac : float, default 0.0001
+        Holes smaller than this fraction of the image area will be closed.
+    mask_smoothing_cycles : int, default 0
+        Number of cycles of (2D) morphological closing to apply to the mask
+    new_labels_key : str | None, default None
+        Key to store the new labels in the SpatialData object
+    inplace: bool, default True
+        Whether to store the new labels in the SpatialData object or return the mask.
+        If the mask is saved to the SpatialData object, it will inherit the scale_factors
+        of the image, if present.
+    **kwargs
+        Optional keyword arguments:
```

Comment:
I am strongly against using kwargs. I can easily imagine the user mistyping `felzenszwalb_params` and wondering why it doesn't change the results :D . We need errors for cases like these.